spec/helpers/settings_helper_spec.rb

```diff
@@ -0,0 +1,222 @@
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+require 'ostruct'
+
+describe SettingsHelper, type: :helper do
+  include Capybara::RSpecMatchers
+
+  shared_examples_for 'labelled' do
+    it { is_expected.to have_selector 'label.form--label' }
+  end
+
+  shared_examples_for 'wrapped in container span' do
+    it { is_expected.to have_selector 'span.form--field-container' }
+  end
+
+  shared_examples_for 'not wrapped in container span' do
+    it { is_expected.not_to have_selector 'span.form--field-container' }
+  end
+
+  describe '#setting_select' do
+    before do
+      expect(Setting).to receive(:field).and_return('2')
+    end
+
+    subject(:output) {
+      helper.setting_select :field, [['Popsickle', '1'], ['Jello', '2'], ['Ice Cream', '3']]
+    }
+
+    it_behaves_like 'labelled'
+    it_behaves_like 'wrapped in container span'
+
+    it 'should output element' do
+      expect(output).to have_selector 'select.form--select > option', count: 3
+      expect(output).to have_select 'settings_field', selected: 'Jello'
+    end
+  end
+
+  describe '#setting_multiselect' do
+    before do
+      expect(Setting).to receive(:field).at_least(:once).and_return('1')
+    end
+
+    subject(:output) {
+      helper.setting_multiselect :field, [['Popsickle', '1'], ['Jello', '2'], ['Ice Cream', '3']]
+    }
+
+    it_behaves_like 'wrapped in container span'
+
+    it 'should have three labels' do
+      expect(output).to have_selector 'label.block', count: 3
+    end
+
+    it 'should output element' do
+      expect(output).to have_selector 'input[type="checkbox"].form--check-box', count: 3
+    end
+  end
+
+  describe '#settings_multiselect' do
+    before do
+      expect(Setting).to receive(:field_a).at_least(:once).and_return('2')
+      expect(Setting).to receive(:field_b).at_least(:once).and_return('3')
+    end
+
+    subject(:output) {
+      helper.settings_multiselect [:field_a, :field_b], [
+        ['Popsickle', '1'], ['Jello', '2'], ['Ice Cream', '3']
+      ]
+    }
+
+    it_behaves_like 'not wrapped in container span'
+
+    it 'should output element' do
+      expect(output).to have_selector 'table'
+      expect(output).to have_selector 'input[type="checkbox"].form--check-box'
+
+      expect(output).to have_checked_field 'field_a_2'
+      expect(output).to have_checked_field 'field_b_3'
+    end
+  end
+
+  describe '#setting_text_field' do
+    before do
+      expect(Setting).to receive(:field).and_return('important value')
+    end
+
+    subject(:output) {
+      helper.setting_text_field :field
+    }
+
+    it_behaves_like 'labelled'
+    it_behaves_like 'wrapped in container span'
+
+    it 'should output element' do
+      expect(output).to include %{
+        <input id="settings_field" name="settings[field]" type="text" value="important value" />
```

Comment:
I am not certain if I understand what you want to do and for what I understand, manipulating the contant (which would not be clean) is the only thing that comes to mind.